components/sidebar/sidebar_category_list/sidebar_category_list.tsx

```diff
@@ -64,9 +80,29 @@ export default class SidebarCategoryList extends React.PureComponent<Props, Stat
             this.scrollToChannel(this.props.currentChannel!.id);
         }
 
+        // TODO: Copying over so it doesn't get lost, but we don't have a design for the sidebar on mobile yet
+        // close the LHS on mobile when you change channels
+        // if (this.props.currentChannel.id !== prevProps.currentChannel.id) {
+        //     if (this.closedDirectChannel) {
+        //         this.closedDirectChannel = false;
+        //     } else {
+        //         this.props.actions.close();
+        //     }
```

Comment:
So first, this is commented out as it's for the mobile sidebar, and I'm still not sure what the plan is for that.

Second, this is original code from `sidebar.jsx`:
`if (this.props.currentTeam === prevProps.currentTeam &&
    this.props.currentChannel.id === prevProps.currentChannel.id &&
    !this.channelIdIsDisplayedForProps(this.props.orderedChannelIds, this.props.currentChannel.id) &&
    this.channelIdIsDisplayedForProps(prevProps.orderedChannelIds, this.props.currentChannel.id)
) {
    this.closedDirectChannel = true;
    redirectUserToDefaultTeam();
    return;
}`

It looks like if your last action was to close a direct channel and get redirected to Town Square, it doesn't want to close the LHS on mobile. Are we preserving this logic?

Reply:
It seems like we'd want that logic, but I can't tell what `this.closedDirectChannel` in particular is used for.

Reply:
When a DM autocloses, the code referenced above is called. Calling `redirectUserToDefaultTeam()` will cause `componentDidUpdate()` to be called again. If `this.closedDirectChannel` is false, it autocloses the mobile sidebar as well.

I assume this logic is so the mobile sidebar autocloses when a user is kicked/removed from a channel, but it doesn't when a DM is autoclosed (so that presumably it doesn't look like your sidebar just closed for no reason)

Reply:
The logic doesn't do anything as written here though. If `this.closedDirectChannel` is true (which never happens), it's set to false, but otherwise, it closes the DM. I think I understand the purpose of this block, but that field seems like it's not needed.

Reply:
Okay I will remove it.

Reply:
Upon further investigation, will use the existing logic.